### ISSUE TITLE
Unknown MySQL server host 'mariadb'

### ISSUE THREAD

**Issue by Toobian** [hardware/mailserver#14]
I can't send or receive mails. When I check logs, I see :
```
2016-04-08T22:13:30.656856+02:00 mail postfix/smtpd[283]: NOQUEUE: reject: RCPT from unknown[***.***.***.***]: 451 4.3.0 <*****@bounce.twitter.com>: Temporary lookup failure; from=<******@bounce.twitter.com> to=<***@***.net> proto=ESMTP helo=<spring-chicken-af.twitter.com>
```
But when i look more i see that :
```
2016-04-08T22:13:30.562427+02:00 mail postfix/trivial-rewrite[287]: warning: connect to mysql server mariadb: Unknown MySQL server host 'mariadb' (2)
2016-04-08T22:13:30.563026+02:00 mail postfix/trivial-rewrite[287]: warning: mysql:/etc/postfix/mysql/mysql-virtual-alias-maps.cf: table lookup problem
2016-04-08T22:13:30.563156+02:00 mail postfix/trivial-rewrite[287]: warning: virtual_alias_domains lookup failure
2016-04-08T22:13:30.655698+02:00 mail postfix/trivial-rewrite[287]: warning: mysql:/etc/postfix/mysql/mysql-virtual-alias-maps.cf: table lookup problem
2016-04-08T22:13:30.656295+02:00 mail postfix/trivial-rewrite[287]: warning: virtual_alias_domains lookup failure
```
Infortunatly, i can't explain why :/ When i try with telnet (from mailserver container to mariadb container) it's working
```
root@mail:/# telnet mariadb 3306   
Trying 172.21.0.2...
Connected to mariadb.
Escape character is '^]'.
b
5.5.5-10.1.13-MariaDB-1~jessieTNlS;n:?�Ov%OA99Srg/Qmysql_native_passwordConnection closed by foreign host.
```

Some idea to help me ?
Thanks :-)

**Comment by Wonderfall** [comment]
May you provide us with your docker-compose file?

**Comment by Toobian** [comment]
Sorry, i missed ><
```
version: "2"

services:
  mailserver:
    image: hardware/mailserver
    container_name: mailserver
    domainname: ***.net
    hostname: mail
    links:
      - mariadb:mariadb
    ports:
      - "25:25"
      - "143:143"
      - "465:465"
      - "587:587"
      - "993:993"
      - "4190:4190"
    environment:
      - OPENDKIM_KEY_LENGTH=4096
      - ADD_DOMAINS=***.net,***.fr
      - DBHOST=mariadb
      - DBUSER=postfix
      - DBNAME=postfix
      - DBPASS=***
    volumes:
      - /docker/mail:/var/mail
      - /docker/dovecot:/var/lib/dovecot
      - /docker/opendkim:/etc/opendkim/keys
      - /etc/localtime:/etc/localtime:ro
      - /docker/proxy/certs/mail.***.net:/etc/letsencrypt/live/mail.***.net
    networks:
      - mail-db

  postfixadmin:
    image: hardware/postfixadmin
    container_name: postfixadmin
    domainname: ***.net
    hostname: mail
    links:
      - mariadb:mariadb
    environment:
      - VIRTUAL_HOST=***.***.net
      - VIRTUAL_NETWORK=nginx-proxy
      - VIRTUAL_PORT=80
      - LETSENCRYPT_HOST=***.***.net
      - LETSENCRYPT_EMAIL=toobian@***.net
      - DBHOST=mariadb
      - DBUSER=postfix
      - DBNAME=postfix
      - DBPASS=***
    volumes:
      - /etc/localtime:/etc/localtime:ro
    networks:
      - proxy-tier
      - mail-db
  rainloop:
    image: wonderfall/rainloop
    container_name: rainloop
    links:
      - mailserver:mailserver
      - mariadb:mariadb
    environment:
      - VIRTUAL_HOST=webmail.***.net
      - LETSENCRYPT_HOST=webmail.***.net
      - LETSENCRYPT_EMAIL=toobian@***.net
      - VIRTUAL_NETWORK=nginx-proxy
      - VIRTUAL_PORT=80
      - GID=991
      - UID=991
    volumes:
      - /docker/rainloop:/rainloop/data
      - /etc/localtime:/etc/localtime:ro
    networks:
      - proxy-tier
      - mail-db

  mariadb:
    image: mariadb:10.1
    container_name: mariadb
    volumes:
      - /docker/mysql/db:/var/lib/mysql
      - /etc/localtime:/etc/localtime:ro
    environment:
      - MYSQL_ROOT_PASSWORD=***
      - MYSQL_DATABASE=postfix
      - MYSQL_USER=postfix
      - MYSQL_PASSWORD=***
    networks:
      - mail-db

networks:
  proxy-tier:
    external:
      name: nginx-proxy
  mail-db:
    external:
      name: mail-db

```

**Comment by Wonderfall** [comment]
Okay, this is interesting because I had similar troubles using dockerfiles second version.
Can't help you, I surrendered!

**Comment by hardware** [comment]
Postfix is isolated with unix chroot mecanism, so check that the mariadb hostname exist in this file : `/var/spool/postfix/etc/hosts` for DNS resolution works properly.

**Comment by Toobian** [comment]
Ok it's working when i add mariadb hostname and its ip address manually.
How can i add hostname automaticly ?

**Comment by hardware** [comment]
It is already the case here : https://github.com/hardware/mailserver/blob/master/rootfs/usr/local/bin/postfix#L7

**Comment by Toobian** [comment]
when i check /etc/hosts file, mariadb hostname is missing. For now i just add hostname in /var/spool/postfix/etc/hosts
If i restart container /etc/hosts will be reset, so i loose my modification.

**Comment by hardware** [comment]
check your docker-compose.yml file syntax, try with version 1.

**Comment by hardware** [comment]
Fix in c2c8a22

Can you confirm ?

**Comment by chmanie** [comment]
Hi, I found this issue while struggling with this in my own set up. Thank you for building this, it's been a great help so far!

I have a question regarding this:

Would it be a problem to run the postfix `rewrite` service in a non-chroot environment?

I changed the line in postfix' `master.cf` to

```
# service type  private unpriv  chroot  wakeup  maxproc command + args
rewrite   unix  -       -       n       -       -       trivial-rewrite
```

and it did work. Will there be any shortcomings or disadvantages due to this change?
